pkg/apply/apply.go

```diff
@@ -0,0 +1,65 @@
+package apply
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ApplyObject applies the desired object against the apiserver,
+// merging it with any existing objects if already present.
+func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) error {
```

Comment:
Can we programatically call Cobra cmds from https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/apply/apply.go so we re-use the code here ? 

This apply add a lot of complexity to the project that we have to maintain.